### PR TITLE
learner is able to rate KC after first satisfaction.

### PR DESCRIPTION
**Pull request by lukaDoric**
## Change goal
- Learner is able to rate KC after first satisfaction.

## UI

image
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,8 @@
     "@auth0/angular-jwt": "~5.1.0",
     "@swimlane/ngx-charts": "~20.1.2",
     "@types/file-saver": "^2.0.5",
+    "angular-star-rating": "^6.0.0",
+    "css-star-rating": "^1.3.1",
```
Comment (Luburic) [inline package.json:33]: This is required by angular star rating or is it an older version that has been abandoned (and should be removed as a result from package.json)?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -59,7 +60,8 @@ export function markdownConfiguration(): MarkedOptions {
     MonitoringModule,
     KnowledgeAnalyticsModule,
     ManagementModule,
-    AuthoringModule
+    AuthoringModule,
+    StarRatingModule.forRoot()
```
Comment (Luburic) [inline src/app/app.module.ts:64]: Should this be in the app module or the learning module?

File: src/app/modules/learning/knowledge-component/kc-rate/kc-rate.component.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<p>Veština savladana, bravo 🥳! Možeš da pređeš na sledeću (klik na taster "Lekcija") ili da pogledaš moj komentar na zadatak (klik na taster "Zatvori").
```
Comment (Luburic) [inline src/app/modules/learning/knowledge-component/kc-rate/kc-rate.component.html:1]: This should be changed following the list discussed on slack.

File: src/app/modules/learning/knowledge-component/kc-rate/kc-rate.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { KcRateService } from './kc-rate.service';
+
+describe('KcRateService', () => {
```
Comment (Luburic) [inline src/app/modules/learning/knowledge-component/kc-rate/kc-rate.service.spec.ts:5]: Remove file.

File: src/app/modules/learning/learning.module.ts [hunk 5]
```diff
@@ -49,21 +52,24 @@ import { TypingAnimatorDirective } from './knowledge-component/submission-result
...
+        YouTubePlayerModule,
+        FlexLayoutModule,
+        RouterModule,
+        FormsModule,
+        MaterialModule,
+        DragDropModule,
+        ReactiveFormsModule,
+        LearningUtilitiesModule,
+        StarRatingModule,
+        MatChipsModule,
```
Comment (Luburic) [inline src/app/modules/learning/learning.module.ts:71]: MatChipsModule should be part of MaterialModule.

File: src/assets/images/star-rating.icons.svg
Comment (Luburic) [inline src/assets/images/star-rating.icons.svg:1]: Do we need this?

File: src/styles.scss [hunk 6]
```diff
@@ -1,4 +1,6 @@
 @import "./cc-primary-theme.scss"; /* You can add global styles to this file, and also import other style files */
+@import '~css-star-rating/scss/star-rating';
```
Comment (Luburic) [inline src/styles.scss:2]: Do we need this?

**Review by Luburic — CHANGES_REQUESTED** [review]
A few minor things to improve along with the slack comments.

File: src/styles.scss (hunk 6 above)
Comment (lukaDoric) [reply src/styles.scss:2]: Yes - https://github.com/BioPhoton/angular-star-rating/wiki/Installing

**Review by Luburic — APPROVED** [review]
Looks good.

**Pull request merged**